### ISSUE TITLE
在fine-tune开始时，程序提示没有tokenizer.json对训练结果有影响吗？

### ISSUE THREAD

**Issue by Ultraman-Orb** [thu-coai/CDial-GPT#48]

**Comment by lemon234071** [comment]
我这里没有看到该提示，您的库版本是不是不太一致？

**Comment by Ultraman-Orb** [comment]
> 我这里没有看到该提示，您的库版本是不是不太一致？
这个不是报错，他是在程序开始加载模型和配置文件时，出现的文字（程序是能运行的），在config\cgpt里确实没有tokenizer.json文件。我跑其它程序时里面都带这个train_tokenized.txt或json形式的文件，这个文件里面是把字转换成数（101 1298 6858 4638 3862 7599 4696 4638 3221 2806 679 857 102 6820 1962 1568 8024 2923 2676 2692 4638 8024 1736 1300 1736 4638 3862 7599 2798 5653 1788 8024 3300 1069 6637 1377 809 1343 1736 1300 1736 1430 1430 3862 7599 ）

**Comment by lemon234071** [comment]
请贴上相关提示，或者截图。我的代码和对应的库应该是不会出现该字段的，我没有检索到。

**Comment by Ultraman-Orb** [comment]
> 请贴上相关提示，或者截图。我的代码和对应的库应该是不会出现该字段的，我没有检索到。

![屏幕截图 2021-01-08 122243](https://user-images.githubusercontent.com/56676402/103974319-8c991700-51ac-11eb-86e4-940defdb7c81.png)
![屏幕截图 2021-01-08 122333](https://user-images.githubusercontent.com/56676402/103974334-94f15200-51ac-11eb-9cb8-6f58a1cbd110.png)
transformers版本没变还是2.1.1 torch 1.7

**Comment by Ultraman-Orb** [comment]
![屏幕截图 2021-01-08 122243](https://user-images.githubusercontent.com/56676402/103975267-af2c2f80-51ae-11eb-91ec-66231a7ca21e.png)
![屏幕截图 2021-01-08 122333](https://user-images.githubusercontent.com/56676402/103975285-b94e2e00-51ae-11eb-8de2-926cc51122ff.png)

**Comment by Ultraman-Orb** [comment]
> 请贴上相关提示，或者截图。我的代码和对应的库应该是不会出现该字段的，我没有检索到。

能否提供一下邮箱？不知道为什么图片始终上传失败

**Comment by lemon234071** [comment]
论文里贴邮箱了呀，而且你可以贴text呀， 例如这样：

    INFO:train.py:Prepare tokenizer, pretrained model and optimizer - add special tokens for fine-tuning

**Comment by Ultraman-Orb** [comment]
> 论文里给邮箱了呀，而且你可以贴text呀， 例如这样：
> 
> ```
> INFO:train.py:Prepare tokenizer, pretrained model and optimizer - add special tokens for fine-tuning
> ```

INFO:transformers.tokenization_utils:Didn't find file config/cgpt/added_tokens.json. We
 won't load it.
INFO:transformers.tokenization_utils:Didn't find file config/cgpt/special_tokens_map.js
on. We won't load it.
INFO:transformers.tokenization_utils:Didn't find file config/cgpt/tokenizer_config.json
. We won't load it.
INFO:transformers.tokenization_utils:loading file config/cgpt/vocab.txt
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.configuration_utils:loading configuration file config/cgpt/config.jso
n

**Comment by Ultraman-Orb** [comment]
> INFO:transformers.tokenization_utils:Didn't find file config/cgpt/added_tokens.json. We
> won't load it.
> INFO:transformers.tokenization_utils:Didn't find file config/cgpt/special_tokens_map.js
> on. We won't load it.
> INFO:transformers.tokenization_utils:Didn't find file config/cgpt/tokenizer_config.json
> . We won't load it.
> INFO:transformers.tokenization_utils:loading file config/cgpt/vocab.txt
> INFO:transformers.tokenization_utils:loading file None
> INFO:transformers.tokenization_utils:loading file None
> INFO:transformers.tokenization_utils:loading file None

**Comment by lemon234071** [comment]
你。。。这是按readme 里quickstart 的操作微调的吗？ 你为啥把config/cgpt指定成模型目录了呢？

**Comment by Ultraman-Orb** [comment]
> 你。。。这是按readme 里quickstart 的操作微调的吗？ 你为啥把config/cgpt指定成模型目录了呢？

这次我严格按照readme文件做的，用git下载了数据集和预训练模型还是有这样的提示
操作指令是D:\python\CDial-GPT-master>python train.py --pretrained --model_checkpoint CDial-
GPT_LCCC-large --data_path data/STC.json --train_batch_size 8 --scheduler linear打开那个云网址后查看了文件确实没有added_tokens.json special_tokens_map.json tokenizer_config.json这几个文件

WARNING:train.py:Running process -1
INFO:train.py:Arguments: Namespace(data_path='data/STC.json', dataset_cache='dataset_ca
che', device='cuda', eval_before_start=False, fp16='', from_step=-1, gpt2=False, gradie
nt_accumulation_steps=64, local_rank=-1, log_file='runs/training.log', lr=5e-05, max_hi
story=15, max_norm=1.0, model_checkpoint='CDial-GPT_LCCC-large', n_emd=768, n_epochs=70
, num_workers=0, pretrained=True, scheduler='linear', train_batch_size=8, valid_batch_s
ize=2, valid_path='data/toy_valid.txt', valid_steps=5000, warmup_steps=5000)
INFO:train.py:Prepare tokenizer, pretrained model and optimizer - add special tokens fo
r fine-tuning
加载预训练模型
INFO:transformers.tokenization_utils:Model name 'CDial-GPT_LCCC-large' not found in mod
el shortcut name list (bert-base-uncased, bert-large-uncased, bert-base-cased, bert-lar
ge-cased, bert-base-multilingual-uncased, bert-base-multilingual-cased, bert-base-chine
se, bert-base-german-cased, bert-large-uncased-whole-word-masking, bert-large-cased-who
le-word-masking, bert-large-uncased-whole-word-masking-finetuned-squad, bert-large-case
d-whole-word-masking-finetuned-squad, bert-base-cased-finetuned-mrpc, bert-base-german-
dbmdz-cased, bert-base-german-dbmdz-uncased). Assuming 'CDial-GPT_LCCC-large' is a path
 or url to a directory containing tokenizer files.
INFO:transformers.tokenization_utils:Didn't find file CDial-GPT_LCCC-large\added_tokens
.json. We won't load it.
INFO:transformers.tokenization_utils:Didn't find file CDial-GPT_LCCC-large\special_toke
ns_map.json. We won't load it.
INFO:transformers.tokenization_utils:Didn't find file CDial-GPT_LCCC-large\tokenizer_co
nfig.json. We won't load it.
INFO:transformers.tokenization_utils:loading file CDial-GPT_LCCC-large\vocab.txt
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.tokenization_utils:loading file None
INFO:transformers.configuration_utils:loading configuration file CDial-GPT_LCCC-large\c
onfig.json
INFO:transformers.configuration_utils:Model config {
  "afn": "gelu",
  "attn_pdrop": 0.1,
  "embd_pdrop": 0.1,
  "finetuning_task": null,
  "id2label": {
    "0": "LABEL_0",
    "1": "LABEL_1"
  },
  "initializer_range": 0.02,
  "is_decoder": false,
  "label2id": {
    "LABEL_0": 0,
    "LABEL_1": 1
  },
  "layer_norm_epsilon": 1e-05,
  "n_ctx": 512,
  "n_embd": 768,
  "n_head": 12,
  "n_layer": 12,
  "n_positions": 513,
  "n_special": 0,
  "num_labels": 2,
  "output_attentions": false,
  "output_hidden_states": false,
  "output_past": true,
  "predict_special_tokens": true,
  "pruned_heads": {},
  "resid_pdrop": 0.1,
  "summary_activation": null,
  "summary_first_dropout": 0.1,
  "summary_proj_to_labels": true,
  "summary_type": "cls_index",
  "summary_use_proj": true,
  "torchscript": false,
  "use_bfloat16": false,
  "vocab_size": 13088
}

INFO:transformers.modeling_utils:loading weights file CDial-GPT_LCCC-large\pytorch_mode
l.bin
INFO:train.py:Prepare datasets
INFO:train.py:Build train and validation dataloaders
INFO:train.py:Load tokenized dataset from cache at dataset_cache_BertTokenizer
D:\anaconda\envs\ysy\lib\site-packages\ignite\handlers\checkpoint.py:708: UserWarning:
Argument save_interval is deprecated and should be None. This argument will be removed
in 0.5.0.Please, use events filtering instead, e.g. Events.ITERATION_STARTED(every=1000
)
  warnings.warn(msg)
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=70.

**Comment by lemon234071** [comment]
忽略即可，用不到这几个文件

**Comment by lemon234071** [comment]
另外 你这个信息里面也没包含tokenizer.json这么个文件啊？？这是啥

**Comment by Ultraman-Orb** [comment]
> 另外 你这个信息里面也没包含tokenizer.json这么个文件啊？？这是啥

我原本想说的就是这些文件，added_tokens.json special_tokens_map.json tokenizer_config.json，不好意思我省略了这么多，我原本就是想问，提示没有这些文件对微调有影响吗？

**Comment by lemon234071** [comment]
要看你下游要做什么任务，我的任务是没有的。 具体你可以去看看huggingface代码看看这些文件干什么的。
